ci: fix hardhat workflow

antonbaliasnikov opened the pull request:
# What ❔

Fixes for hardhat workflow:
* [x] Downgrade hardhat to 2.20 to fix execution issue
* [x] Hacks for `zksafe` project 

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

To stabilize hardhat testing.

<!-- Why are these changes done? What goal do they contribute to? What are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [x] PR title corresponds to the body of PR.
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

hedgar2017 reviewed (changes requested):
It looks like the Hardhat report was not updated here.
zksafe is showing zeros.

antonbaliasnikov commented:
> It looks like the Hardhat report was not updated here. zksafe is showing zeros.

@hedgar2017 , thanks for noticing. I've fixed the run error on `solc`, all tests are passing now, but with `solx` we're getting an error:
```
  ZkSafeModule
The Hardhat Network tracing engine could not be initialized. Run Hardhat with --verbose to learn more.
    1) "before all" hook for "Should succeed verification of a basic transaction"


  0 passing (760ms)
  1 failing

  1) ZkSafeModule
       "before all" hook for "Should succeed verification of a basic transaction":
     HardhatPluginError: The deployment wasn't successful, there were failures:

Failures:

  * zkSafe_v1#HonkVerifier/1: Simulating the transaction failed with error: Reverted without reason
      at Proxy.deploy (/Users/abalias/projects/zksafe/node_modules/.pnpm/@nomicfoundation+hardhat-ignition-viem@0.15.11_@nomicfoundation+hardhat-ignition@0.15.1_5d4d80f66af0b8f0c365be2d404d3825/node_modules/@nomicfoundation/hardhat-ignition-viem/src/viem-ignition-helper.ts:167:13)
      at processTicksAndRejections (node:internal/process/task_queues:105:5)
      at async Context.<anonymous> (/Users/abalias/projects/zksafe/test/ZkSafeModule.ts:85:24)



 ELIFECYCLE  Test failed. See above for more details.
```
